pygfx/renderers/wgpu/meshshader.py

```diff
@@ -68,9 +68,18 @@ def get_bindings(self, wobject, shared):
         if getattr(geometry, "normals", None) is not None:
             normal_buffer = geometry.normals
         else:
-            normal_data = normals_from_vertices(
-                geometry.positions.data, geometry.indices.data
-            )
+            if geometry.indices.data.shape[1] == 4:
+                n1 = normals_from_vertices(
+                    geometry.positions.data, geometry.indices.data[:,:3]
+                )
+                n2 = normals_from_vertices(
+                    geometry.positions.data, geometry.indices.data[:,[0,2,3]]
+                )
+                normal_data = (n1+n2)/2
```

Comment:
I think this should do the trick; first you triangulate all the quads, then you compute the vertex normals based on the triangles in a single step. Now the weighted sum will be correct.

```suggestion
                triangle_indices = np.concatenate([
                    geometry.indices.data[:, :3],
                    geometry.indices.data[:, [0,2,3]],
                ], axis=0)
                normal_data = normals_from_vertices(
                    geometry.positions.data,
                    triangle_indices,
                )
```